### PR TITLE
add setLocation function for fbsimctl

### PR DESCRIPTION
**Pull request by noshuai**
add setLocation function for fbsimctl, such as:
./fbsimctl xx-xxxx-xxxx-xxx-xxxxx setLocation 46 116

### CODE REVIEW

**Comment by facebook-github-bot-5** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

File: .gitignore [hunk 1]
```diff
@@ -45,3 +45,11 @@ Carthage/Checkouts
 fbsimctl/Carthage/Build
 fbsimctl/Carthage/Checkouts
 
+
```
Comment (lawrencelomax) [inline .gitignore]: I'm not sure why you'd want to add this to the `gitignore`. Please remove.

File: FBSimulatorControl/Interactions/FBSimulatorInteraction+Bridge.h [hunk 2]
```diff
@@ -43,4 +43,7 @@
  */
 - (instancetype)tap:(double)x y:(double)y;
 
+
+- (instancetype)setLocation:(double)x y:(double)y;
```
Comment (lawrencelomax) [inline FBSimulatorControl/Interactions/FBSimulatorInteraction+Bridge.h]: `x` and `y` are poorly named in this context. Use `longitude` and `latitude` throughout.

File: fbsimctl/FBSimulatorControlKit/Sources/Command.swift [hunk 3]
```diff
@@ -83,6 +83,7 @@ public enum Action {
   case Uninstall(String)
   case Upload([FBDiagnostic])
   case WatchdogOverride([String], NSTimeInterval)
+  case setLocation(Double,Double)
```
Comment (lawrencelomax) [inline fbsimctl/FBSimulatorControlKit/Sources/Command.swift]: While newer versions of swift don't have capitalization of their first character, this is inconsistent with the rest of the Actions. When we do migrate to using the new syntax, we can do it all at once. Please make the first character uppercase to be consistent with the rest of the actions.

File: fbsimctl/FBSimulatorControlKit/Sources/Command.swift [hunk 4]
```diff
@@ -216,6 +217,8 @@ public func == (left: Action, right: Action) -> Bool {
     return true
   case (.Tap(let leftX, let leftY), .Tap(let rightX, let rightY)):
     return leftX == rightX && leftY == rightY
+  case (.setLocation(let leftX, let leftY), .setLocation(let rightX, let rightY)):
```
Comment (lawrencelomax) [inline fbsimctl/FBSimulatorControlKit/Sources/Command.swift]: Let's use `longitude` and `latitude`

File: fbsimctl/FBSimulatorControlKit/Sources/Events.swift [hunk 5]
```diff
@@ -43,6 +43,7 @@ public enum EventName : String {
   case Uninstall = "uninstall"
   case Upload = "upload"
   case WatchdogOverride = "watchdog_override"
+  case setLocation = "setLocation"
```
Comment (lawrencelomax) [inline fbsimctl/FBSimulatorControlKit/Sources/Events.swift]: Camel casing is inconsistent with the naming of the other APIs. Use `set_location` as this matches the `watchdog_override` naming.

File: fbsimctl/FBSimulatorControlKit/Sources/HttpRelay.swift [hunk 6]
```diff
@@ -200,6 +200,13 @@ class HttpRelay : Relay {
       return Action.Tap(x, y)
     }
   }}
+  private var setLocationRoute: HttpRoute { get {
+        return HttpRoute(method: HttpMethod.POST, endpoint: EventName.setLocation.rawValue) { json in
+            let x = try json.getValue("x").getNumber().doubleValue
```
Comment (lawrencelomax) [inline fbsimctl/FBSimulatorControlKit/Sources/HttpRelay.swift]: `longitude` and `latitude`

File: fbsimctl/FBSimulatorControlKit/Sources/HttpRelay.swift [hunk 7]
```diff
@@ -200,6 +200,13 @@ class HttpRelay : Relay {
...
+            let y = try json.getValue("y").getNumber().doubleValue
+            return Action.setLocation(x, y)
+        }
+        }}
```
Comment (lawrencelomax) [inline fbsimctl/FBSimulatorControlKit/Sources/HttpRelay.swift]: This indentation needs to be consistent with the other route methods

File: fbsimctl/FBSimulatorControlKit/Sources/HttpRelay.swift [hunk 8]
```diff
@@ -200,6 +200,13 @@ class HttpRelay : Relay {
       return Action.Tap(x, y)
     }
   }}
+  private var setLocationRoute: HttpRoute { get {
```
Comment (lawrencelomax) [inline fbsimctl/FBSimulatorControlKit/Sources/HttpRelay.swift]: Need a newline above here.

File: fbsimctl/FBSimulatorControlKit/Sources/SimulatorRunners.swift [hunk 9]
```diff
@@ -127,6 +127,10 @@ struct SimulatorActionRunner : Runner {
       return SimulatorInteractionRunner(reporter, EventName.Tap, ControlCoreSubject(simulator)) { interaction in
         interaction.tap(x, y: y)
       }
+    case .setLocation(let x, let y):
```
Comment (lawrencelomax) [inline fbsimctl/FBSimulatorControlKit/Sources/SimulatorRunners.swift]: `longitude` `latitude`

**Comment by lawrencelomax** [comment]
Logic looks sounds, mostly indentation and naming that needs fixing.

File: FBSimulatorControl/Interactions/FBSimulatorInteraction+Bridge.m [hunk 10]
```diff
@@ -43,6 +43,15 @@ - (instancetype)tap:(double)x y:(double)y
...
 
+- (instancetype)setLocation:(double)latitude longitude:(double)longitude
+{
+    return [self interactWithBridge:^ BOOL (NSError **error, FBSimulator *simulator, FBSimulatorBridge *bridge) {
+        [bridge setLocationWithLatitude:latitude longitude:longitude];
+        return YES;
+    }];
+}
+
+
```
Comment (lawrencelomax) [inline FBSimulatorControl/Interactions/FBSimulatorInteraction+Bridge.m]: nit: uncessary double linebreak

File: fbsimctl/FBSimulatorControlKit/Sources/Command.swift [hunk 11]
```diff
@@ -83,6 +83,7 @@ public enum Action {
   case Uninstall(String)
   case Upload([FBDiagnostic])
   case WatchdogOverride([String], NSTimeInterval)
+  case SetLocation(Double,Double)
```
Comment (lawrencelomax) [inline fbsimctl/FBSimulatorControlKit/Sources/Command.swift]: nit: space after the comma: `SetLocation(Double, Double)`

File: fbsimctl/FBSimulatorControlKit/Sources/Command.swift [hunk 12]
```diff
@@ -280,6 +283,8 @@ extension Action {
       return (EventName.Diagnose, nil)
     case .WatchdogOverride(let bundleIDs, _):
       return (EventName.WatchdogOverride, ArraySubject(bundleIDs))
+    case .setLocation:
```
Comment (lawrencelomax) [inline fbsimctl/FBSimulatorControlKit/Sources/Command.swift]: This should be uppercase like the declaration of your enum. This is why you are [getting build errors on CI](https://travis-ci.org/facebook/FBSimulatorControl/jobs/149380233#L1823)

File: fbsimctl/FBSimulatorControlKit/Sources/Events.swift [hunk 13]
```diff
@@ -43,6 +43,7 @@ public enum EventName : String {
   case Uninstall = "uninstall"
   case Upload = "upload"
   case WatchdogOverride = "watchdog_override"
+  case setLocation = "set_location"
```
Comment (lawrencelomax) [inline fbsimctl/FBSimulatorControlKit/Sources/Events.swift]: Make first letter uppercase to be consistent with everywhere else

File: fbsimctl/FBSimulatorControlKit/Sources/HttpRelay.swift [hunk 14]
```diff
@@ -200,6 +200,14 @@ class HttpRelay : Relay {
...
     }
   }}
+    
+  private var setLocationRoute: HttpRoute { get {
+    return HttpRoute(method: HttpMethod.POST, endpoint: EventName.setLocation.rawValue) { json in
+      let latitude = try json.getValue("latitude").getNumber().doubleValue
+      let longitude = try json.getValue("longitude").getNumber().doubleValue
+      return Action.setLocation(latitude, longitude)
+        }
+        }}
```
Comment (lawrencelomax) [inline fbsimctl/FBSimulatorControlKit/Sources/HttpRelay.swift]: Indentation of the `}}` should be at the same level as `private` on line 204

File: fbsimctl/FBSimulatorControlKit/Sources/HttpRelay.swift [hunk 15]
```diff
@@ -200,6 +200,14 @@ class HttpRelay : Relay {
       return Action.Tap(x, y)
     }
   }}
+    
+  private var setLocationRoute: HttpRoute { get {
+    return HttpRoute(method: HttpMethod.POST, endpoint: EventName.setLocation.rawValue) { json in
+      let latitude = try json.getValue("latitude").getNumber().doubleValue
+      let longitude = try json.getValue("longitude").getNumber().doubleValue
+      return Action.setLocation(latitude, longitude)
+        }
```
Comment (lawrencelomax) [inline fbsimctl/FBSimulatorControlKit/Sources/HttpRelay.swift]: Indentation of `}` should be at the same level as `return` on line 205

File: .gitignore [hunk 16]
```diff
@@ -44,4 +44,3 @@ Carthage/Build
 Carthage/Checkouts
 fbsimctl/Carthage/Build
 fbsimctl/Carthage/Checkouts
-
```
Comment (lawrencelomax) [inline .gitignore]: This line removal is unnecessary in your commits.

**Pull request closed without merging**